src/olympia/users/tasks.py

```diff
@@ -112,3 +122,144 @@ def sync_blocked_emails(batch_size=BATCH_SIZE, **kw):
                 SuppressedEmail.objects.bulk_create(email_blocks, ignore_conflicts=True)
+
+
+@task(autoretry_for=(HTTPError, Timeout), max_retries=5, retry_backoff=True)
+def send_suppressed_email_confirmation(suppressed_email_verification_id):
+    verification = SuppressedEmailVerification.objects.filter(
+        id=suppressed_email_verification_id
+    ).first()
+
+    if not verification:
+        raise Exception(f'invalid id: {suppressed_email_verification_id}')
+
+    email = verification.suppressed_email.email
+
+    url = (
+        f'{settings.SOCKET_LABS_HOST}servers/{settings.SOCKET_LABS_SERVER_ID}/'
+        f'suppressions/remove?emailAddress={email}'
+    )
+    headers = {
+        'authorization': f'Bearer {settings.SOCKET_LABS_TOKEN}',
+    }
+
+    try:
+        task_log.info(f'removing email: {email} from suppression list')
+        response = requests.delete(url, headers=headers)
+        response.raise_for_status()
+    except HTTPError as http_err:
+        if response.status_code == 404:
+            task_log.warn(f'email not found in suppression list: {email}')
+            pass
+        else:
+            raise http_err
+
+    code_snippet = str(verification.confirmation_code)[-5:]
+
+    verification.status = SuppressedEmailVerification.STATUS_CHOICES.Pending
+
+    confirmation_link = (
+        # TODO: replace with email-verification reverse path
+        '' + '?code=' + str(verification.confirmation_code)
+    )
+
+    send_mail_jinja(
+        gettext(f'Verify your email ({code_snippet})'),
+        'devhub/emails/verify-email-requested.ltxt',
+        {
+            'confirmation_link': confirmation_link,
+        },
+        recipient_list=[email],
+    )
+
+    verification.save()
+    check_suppressed_email_confirmation.delay(verification.id)
+
+
+@task(
+    autoretry_for=(
+        HTTPError,
+        Timeout,
+    ),
+    max_retries=5,
+    retry_backoff=True,
+)
+def check_suppressed_email_confirmation(suppressed_email_verification_id, page_size=5):
+    verification = SuppressedEmailVerification.objects.filter(
+        id=suppressed_email_verification_id
+    ).first()
+
+    if not verification:
+        raise Exception(f'invalid id: {suppressed_email_verification_id}')
+
+    email = verification.suppressed_email.email
+
+    current_count = 0
+    total = 0
+
+    code_snippet = str(verification.confirmation_code)[-5:]
+    path = f'servers/{settings.SOCKET_LABS_SERVER_ID}/reports/recipient-search/'
+
+    # socketlabs might set the queued time any time between 00:00 and 23:59
+    # so we need to check the full day before creation
+    # and to the end of the current day
+    start_date = (verification.created - datetime.timedelta(days=1)).replace(
+        hour=0, minute=0, second=0, microsecond=0
+    )
+    end_date = datetime.datetime.now().replace(
+        hour=23, minute=59, second=59, microsecond=999999
+    )
+    date_format = '%Y-%m-%dT%H:%M:%S%z+0100'
```

Comment:
I will try with this, but I was having trouble getting socketlabs to return a response unless I added the time in this precise format.

Reply:
I will test your proposal in the next PR where we actually trigger this job. It will be much easier to test there as I need to test against the actual socketlabs api.

Reply:
Using iso format returns

```bash
{"error":[{"errorType":"InvalidDateFormat","message":"The requested dates are not in the correct format. Please review the API documentation for allowed formats."}]}%
```

Curl
```bash
curl --location 'https://api.socketlabs.com/v2/servers/2301/reports/recipient-search/?toEmailAddress=kmeinhardt%40mozilla.com&startDate=2024-01-07T00%3A00%3A00&endDate=2024-01-09T08%3A51%3A16.814649&pageNumber=0&pageSize=5&sortField=queuedTime&sortDirection=dsc' \
--header 'Authorization: Bearer <token>'
```

I was actually able to get valid responses for using only y-m-d formatting. So i've added a commit in the next PR to reformat to this as it is simpler. If this works consistently in dev we can keep it simple.